core/kazoo_number_manager/src/carriers/knm_carriers.erl

```diff
@@ -318,20 +318,32 @@ acquire(Number, ?NE_BINARY = Mod, 'false') ->
 -spec disconnect(knm_number:knm_number()) ->
                         knm_number:knm_number().
 disconnect(Number) ->
-    Module = carrier_module(Number),
-    Module:disconnect_number(Number).
+    case knm_phone_number:module_name(knm_number:phone_number(Number)) of
+        ?NE_BINARY=Mod ->
+            Module = wh_util:to_atom(Mod, 'true'),
+            lager:debug("contacting carrier ~s", [Module]),
+            Module:disconnect_number(Number);
+        _Mod ->
+            lager:debug("non-existant carrier module ~p, allowing disconnect", [_Mod]),
+            Number
```

Comment:
knm_carriers:disconnect/1 is only called by knm_number:release which calls into number_states which does the state changing stuff